pilot/proxy/envoy/config.go

```diff
@@ -792,6 +792,15 @@ func appendPortToDomains(domains []string, port int) []string {
 	return domainsWithPorts
 }
 
+func truncateClusterName(name string) string {
```

Comment:
can you add a unit test with boundary checks (ie size just under max, at max, at max + 1)

Reply:
yep!